memgpt/orm/agent.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional, List, TYPE_CHECKING
+from sqlalchemy import String, JSON
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from memgpt.orm.sqlalchemy_base import SqlalchemyBase
+from memgpt.orm.mixins import OrganizationMixin
+from memgpt.data_types import LLMConfig, EmbeddingConfig
+
+if TYPE_CHECKING:
+    from memgpt.orm.organization import Organization
+    from memgpt.orm.source import Source
+    from memgpt.orm.user import User
+    from memgpt.orm.tool import Tool
+
+class Agent(SqlalchemyBase, OrganizationMixin):
+    __tablename__ = 'agent'
+
+    name:Mapped[Optional[str]] = mapped_column(String, nullable=True, doc="a human-readable identifier for an agent, non-unique.")
+    persona: Mapped[str] = mapped_column(doc="the persona text for the agent, current state.")
+    # TODO: reconcile this with persona,human etc AND make this structured via pydantic!
+    # TODO: these are vague and need to be more specific and explained. WTF is state vs _metadata?
```

Comment:
state is the in-process state of the agent (e.g. the memory, in-context message IDs) while metadata is arbitrary info the user might want to attach to the agent